cpp/daal/src/threading/threading.h

```diff
@@ -480,10 +617,23 @@ class static_tls
     static_tls_deleter * _deleter    = nullptr;
 };
 
+/// Local storage (LS) for a data of a thread.
+/// Does not change its local variable after nested parallel constructs,
+/// but can have performance penalties comparing to daal::tls.
+/// Can be safely used in case of nested parallel regions.
+///
+/// @tparam F  Type of the data located in the storage
 template <typename F>
 class ls : public tlsBase
 {
 public:
+    /// Initialize a local storage.
+    ///
+    /// @tparam lambdaType  Lambda function of type [/* captures */]() -> F
+    ///
+    /// @param lambda   Lambda function that initializes a local storage
+    /// @param isTls    if true, then local storage is a thread-local storage (daal::tls)
+    ///                 and might have problems in case of nested parallel regions.
```

Comment:
I think it's worth putting a note somewhere in the documentation about nested parallelism, and what problems that can create for threads. Then from the comments in this file, you can link to the note. Just saying that there might be problems opens up questions, rather than resolving questions.